bentoml/cli/deployment.py

```diff
@@ -147,341 +146,79 @@ def get_state_after_await_action_complete(
-            'replicas': replicas,
-            'service_name': service_name,
-            'service_type': service_type,
-            'memory_size': memory_size,
-            'timeout': timeout,
-        }
-        yatai_client = YataiClient()
+    def deploy(deployment_yaml, output, wait):
+        track_cli('deploy-deploy', deployment_yaml.get('spec', {}).get('operator'))
         try:
-            result = yatai_client.deployment.create(
-                name,
-                namespace,
-                bento_name,
-                bento_version,
-                platform,
-                operator_spec,
-                parse_key_value_pairs(labels),
-                parse_key_value_pairs(annotations),
-            )
-        except BentoMLException as e:
-            _echo(
-                'Failed to create deployment {}.: {}'.format(name, str(e)),
-                CLI_COLOR_ERROR,
-            )
-            return
-
-        if result.status.status_code != status_pb2.Status.OK:
-            _echo(
-                'Failed to create deployment {name}. {error_code}:'
-                '{error_message}'.format(
-                    name=name,
-                    error_code=status_pb2.Status.Code.Name(result.status.status_code),
-                    error_message=result.status.error_message,
-                ),
-                CLI_COLOR_ERROR,
-            )
-            return
-        else:
-            if wait:
-                result_state = get_state_after_await_action_complete(
-                    yatai_client=yatai_client,
-                    name=name,
-                    namespace=namespace,
-                    message='Creating deployment ',
+            yatai_client = YataiClient()
+            result = yatai_client.deployment.apply(deployment_yaml)
+            if result.status.status_code != status_pb2.Status.OK:
+                _echo(
+                    'Failed to deploy deployment {name}. '
+                    '{error_code}:{error_message}'.format(
+                        name=deployment_yaml.get('name'),
+                        error_code=status_pb2.Status.Code.Name(
+                            result.status.status_code
+                        ),
+                        error_message=result.status.error_message,
+                    ),
+                    CLI_COLOR_ERROR,
                 )
-                if result_state.status.status_code != status_pb2.Status.OK:
-                    _echo(
-                        'Created deployment {name}, failed to retrieve latest status.'
-                        ' {error_code}:{error_message}'.format(
-                            name=name,
-                            error_code=status_pb2.Status.Code.Name(
-                                result_state.status.status_code
-                            ),
-                            error_message=result_state.status.error_message,
-                        )
+            else:
+                if wait:
```

Comment:
similarly, this `wait` should go into the `yatai_client.deployment.create` method?

Reply:
lol, didnt update it